src/common/hist_util.h

```diff
@@ -407,7 +407,60 @@ class GHistIndexBlockMatrix {
  *     for that particular bin
  *  Uses global bin id so as to represent all features simultaneously
  */
-using GHistRow = Span<tree::GradStats>;
+
+struct GradStatHist {
```

Comment:
I will stop you from continuing the process. ;-)  Please avoid copying code, use template instead if applicable.  Also why not `GradientPair`?

Reply:
`GradientPair` was used. Thanks  :)